crates/re_ui/src/list_item2/scope.rs

```diff
@@ -13,18 +12,54 @@ pub struct State {
     // be generalized to some `full_span_scope` mechanism to be used by all full-span widgets beyond
     // `ListItem`.
     pub(crate) background_x_range: egui::Rangef,
-    // TODO(ab): record the use of right action button in all PropertyContent such as to not reserve
-    // right gutter space if none have it.
+
+    /// Left-most X coordinate for the scope.
+    ///
+    /// This is the reference point for tracking column width. This is set by [`list_item_scope`]
+    /// based on `ui.max_rect()`.
+    pub(crate) left_x: f32,
+
+    /// Column width to be used this frame.
+    ///
+    /// The column width has `left_x` as reference, so it includes:
+    /// - All the indentation on the left side of the list item.
+    /// - Any extra indentation added by the list item itself.
+    /// - The list item's collapsing triangle, if any.
+    ///
+    /// The effective left column width for a given [`super::ListItemContent`] implementation can be
+    /// calculated as `left_column_width - (context.rect.left() - left_x)`.
+    pub(crate) left_column_width: Option<f32>,
```

Comment:
What does it mean for this to be `None`?

Reply:
On the first frame, the scope doesn't now how to initialise that. So `None` tell ListItemContent to "use their best judgment. For `PropertyContent` it uses 50/50.

Reply:
clarified in the docstring